packages/ns-api/files/ns.threatshield

```diff
@@ -0,0 +1,182 @@
+    for f in feeds:
+        feed = feeds[f]
+        if f.endswith('lvl1'):
+            confidence = 10
+        elif f.endswith('lvl2'):
+            confidence = 8
+        elif f.endswith('lvl3'):
+            confidence = 6
+        else:
+            confidence = -1
+
+        if 'bl.nethesis.it' in feed['url_4']:
+            type = 'enterprise'
+        else:
+            type = 'community'
+        enabled = f in enabled_feeds
+        # show only enterprise lists if the subscription is available
+        if enterprise_subscription and type == 'community':
+            if not enabled:
+                continue
+            else:
+                type = 'warning'
+        ret.append({ 'name': f, 'type': type, 'enabled': enabled, 'confidence': confidence})
+    return { "data": ret }
+
+def list_settings(e_uci):
+    return { 'data': {'enabled': e_uci.get('banip', 'global', 'ban_enabled') == '1' } }
+
+def edit_blocklist(e_uci, payload):
+    feeds = list_feeds()
+    for f in payload['enabled']:
+        if f not in feeds:
+            raise ValidationError('blocklist_name', 'invalid')
+    e_uci.set('banip', 'global', 'ban_feed', payload['enabled'])
+    e_uci.save('banip')
+    return {'message': 'success'}
+
+def edit_settings(e_uci, payload):
+    if payload['enabled']:
+        e_uci.set('banip', 'global', 'ban_enabled', '1')
+    else:
+        e_uci.set('banip', 'global', 'ban_enabled', '0')
+    e_uci.save('banip')
+    return {'message': 'success'}
+
+def list_allowed():
+    return { "data": get_allow_list() }
+
+def add_allowed(payload):
+    cur = get_allow_list()
+    # extract address from cur list
+    if payload['address'] in [x['address'] for x in cur]:
+        raise ValidationError('address', 'address_already_present', payload['address'])
+    cur.append({ "address": payload['address'], "description": payload['description'] })
+    write_allow_list(cur)
+    return {'message': 'success'}
+
+def delete_allowed(payload):
+    cur = get_allow_list()
+    if payload['address'] not in [x['address'] for x in cur]:
+        raise ValidationError('address', 'address_not_found', payload['address'])
+    # remove address from cur list
+    for i in range(len(cur)):
+        if cur[i]['address'] == payload['address']:
+            del cur[i]
+            break
+    write_allow_list(cur)
+    return {'message': 'success'}
+
+def edit_allowed(payload):
+    cur = get_allow_list()
+    if payload['address'] not in [x['address'] for x in cur]:
+        raise ValidationError('address', 'address_not_found', payload['address'])
+    for i in range(len(cur)):
+        if cur[i]['address'] == payload['address']:
+            cur[i]['description'] = payload['description']
+            break
+    write_allow_list(cur)
+    return {'message': 'success'}
+
+cmd = sys.argv[1]
+
+if cmd == 'list':
+    print(json.dumps({
+        'list-blocklist': {},
+        'edit-blocklist': { "enabled": ["blocklist_name", "another_blocklist_name"] },
```

Comment:
The UI doesn't have a "Save" button below blocklist toggles, so it will call `edit-blocklist` anytime the user clicks a toggle. Maybe it's better to use an interface like this:
```
'edit-blocklist': { "blocklist": "blocklist_name", "enabled": True },
```

Reply:
Done